### ISSUE TITLE
Architecture improvement: isolate DB access to the Sentry

### ISSUE THREAD

**Issue by Ivshti** [AdExNetwork/adex-validator-stack-js#78]
possibly replaces the need of #31 

The proposal is to change the architecture in the following way:

- [x] The Sentry is the only component that has a connection to the database
- [x] The worker ticks are refactored into pure, testable functions
- [x] The worker executable (bin/validatorWorker) would retrieve the latest messages from the Sentry, the latest event aggregates, and produce new message (either NewState, ApproveState/InvalidNewState) and submit it back to the Sentry (actually, propagates the messages to everyone)

There are a bunch of benefits:

- Replaces #31 and allows us to swap out the DB more easily
- Allows easier testability by testing the follower/leader tick pure functions
- Security: before producing a new state, we'd always validate the previous one and whether we signed it; DB tampering becomes less of an issue

This, however, also requires *dropping* the Watcher. Instead, that functionality would be moved to the adapter. When a channel is submitted, the adapter will check if it's admissible. For Ethereum, this would mean that the channel is submitted on-chain. This also has the advantage of immediate feedback.

In order to still preserve the ability for end-users to submit channels without waiting for them to be submitted on-chain, we can implement a submission API in the Market or more appropriately the Relayer that would try to submit the channel to the validator, *after* it's on-chain.

Another possible issue is that there's no API to submit the full tree in the Sentry. However, we can introduce an `Accounting` message type to do the same. In order to optimize this for storage/performance in the future, we can purge the tree for the older states.

----------

**TODO:**

- [x] adapter.sign should ensure replay protection itself; see Heartbeat - getSignableStateRoot does this exact thing
- [x] REFACTOR: validator msgs: sequential ID?
- [x] REFACTOR: /last-approved/ method on the sentry AND tests
- [x] REFACTOR: newState: should only use `balances`
- [x] REFACTOR: drop `isValidValidatorFees`; this check is pointless, since ultimately we're signing and enforcing OUTPACE/health on `balancesAfterFees`; so if you want to deceive the follower, you can always forge a preimage of `balancesAfterFees` which will pass the check; in other words, nothing really depends on the balances before fees (in NewState/ApproveState), so no point in enforcing it
- [x] API: should we wrap validator messages in `msg`? we lose `from` and `received`: **decision:** yes, we lose nothing from having that
- [x] sentry interface, in place of the propagation lib
- [x] drop nothingNew - ticks would be ran every time
- [x] drop stateTree
- [x] make everything work
- [x] accounting: { lastEvAggr, balancesBeforeFees, balances }
- [x] InvalidNewState should be emitted only once; so ticks should happen only if there is no OUR msg with that stateRoot
- [x] test: we only respond to a new NewState once
- [x] read through the doc/pseudocode and see if we missed anything
- [x] decide on purging/cleanup logic

------------

and some tests improvement in a separate PR:

- [x] test: DRY, async/await
- [x] test: no NewStates are being produced when we just trigger the tick
- [x] test: invalid transition
- [x] test: propagation works

**Comment by samparsky** [comment]
Use async / await syntax #77 for the refactoring

**Comment by Ivshti** [comment]
https://gist.github.com/Ivshti/dbc41f27e2928e600ab95c57e45782d3

**Comment by Ivshti** [comment]
done with the merge of #88
